### PR TITLE
106 support modified settings25 retrieve field setup by ish type

### PR DESCRIPTION
**Pull request by OlegTokar**
Please take a look at changes for #106 and #58 
Notes:
#58 fix will work for Docs 14.0.4 and greater releases  (because Extension XML settings parsing for previous releases is not implemented)
in #106 please double check agreed breaking changes (image attached to the story) and they are also described in Draft https://github.com/sdl/ISHRemote/releases/
I also modified in PesterSetup.ps1 the way how we load ISHRemote because otherwise it wasn't picking up formatting specified in ISHRemote.Format.ps1xml

### CODE REVIEW

**Review by HildeVermeiren — APPROVED** [review]

File: Source/ISHRemote/Trisoft.ISHRemote/ISHRemote.PesterSetup.ps1 [hunk 1]
```diff
@@ -9,7 +9,7 @@ Write-Host "Generating and Executing Import-Module Statement..."
 $project = (Split-Path -Parent $MyInvocation.MyCommand.Path).ToLower()
 $project = $project.Substring(0, $project.LastIndexOf("ishremote"))
 $project = ($project + "ishremote")
-Import-Module (Join-Path $project "\bin\debug\ishremote.psm1") -DisableNameChecking
+Import-Module (Join-Path $project "\bin\debug\ISHRemote") -DisableNameChecking
```
Comment (ddemeyer) [inline Source/ISHRemote/Trisoft.ISHRemote/ISHRemote.PesterSetup.ps1:12]: Why is this removed, I seem to remember that forcing `.psm1` was required to make sure that from a file path the module is loading over the PS1 file and not directly the DLL

File: Source/ISHRemote/Trisoft.ISHRemote/Objects/Public/IshTypeFieldDefinition.cs [hunk 2]
```diff
@@ -55,31 +61,34 @@ public class IshTypeFieldDefinition : IComparable
...
+                    case Enumerations.DataType.ISHMetadataBinding:
+                        return ReferenceMetadataBinding;
                     case Enumerations.DataType.ISHLov:
                         return ReferenceLov;
                     case Enumerations.DataType.ISHType:
                         // return sorted to avoid string compare issues when database seq differed
                         return string.Join(",",ReferenceType.OrderBy(q => q).ToList());
                     default:
-                        return DataType.ToString();
+                        return String.Empty;
```
Comment (ddemeyer) [inline Source/ISHRemote/Trisoft.ISHRemote/Objects/Public/IshTypeFieldDefinition.cs:91]: This changed because?

Comment (OlegTokar) [reply Source/ISHRemote/Trisoft.ISHRemote/Objects/Public/IshTypeFieldDefinition.cs:91]: Because that is how I read image attached where "date/number/string we see empty". We can discuss this again if there is a value to see "String" for a "String"

File: Source/ISHRemote/Trisoft.ISHRemote/ISHRemote.PesterSetup.ps1 (hunk 1 above)
Comment (OlegTokar) [reply Source/ISHRemote/Trisoft.ISHRemote/ISHRemote.PesterSetup.ps1:12]: @ddemeyer This is how it works for me in Powershell ISE to see objects properly formatted. I am not directly specifying dll file, just "ISHRemote". Can you double check how it works for you?

Comment (ddemeyer) [reply Source/ISHRemote/Trisoft.ISHRemote/ISHRemote.PesterSetup.ps1:12]: Let's continue with your code... hopefully we both remember this if it has side effects :)

File: Source/ISHRemote/Trisoft.ISHRemote/Objects/Public/IshTypeFieldDefinition.cs (hunk 2 above)
Comment (ddemeyer) [reply Source/ISHRemote/Trisoft.ISHRemote/Objects/Public/IshTypeFieldDefinition.cs:91]: In the past, it was never rendered in `Out-GridView` and the likes... I might be a bit confused here
* DataType column, which is filled with Type object property remains as is, besides adding ISHMetadataBinding
* New column DataSource, which is filled by TypeSource can indeed return empty, because it would be superfluous to have String and String next to each other...

Comment (OlegTokar) [reply Source/ISHRemote/Trisoft.ISHRemote/Objects/Public/IshTypeFieldDefinition.cs:91]: Correct, DataType column filled with Type and only gets new ISHMetadataBinding
DataSource is filled with TypeSource(former Type) now returns ReferenceMetadataBinding for ISHMetadataBinding and empty string for all unsupported values (so not ISHLov or ISHMetadataBinding or ISHType). So a second 'breaking' change described in draft of https://github.com/sdl/ISHRemote/releases/

**Review by ddemeyer — APPROVED** [review]
